search/includes/classes/class-search.php

```diff
@@ -840,6 +852,10 @@ private function get_index_exists_option_name( $url ) {
 	 * @return array  $request  New request
 	 */
 	public function filter__ep_do_intercept_request( $request, $query, $args, $failures = 0, $type = null ) {
+		if ( is_wp_error( $request ) && 503 === $request->get_error_code() ) {
```

Comment:
What are your thoughts on putting this logic in a later filter to run after the other ones hooked onto `ep_do_intercept_request` instead of modifying `filter__ep_do_intercept_request` and `get_cached_index_exists_request`?

Reply:
The idea is that when we reach the concurrency limit, we should not make a request to ES.

Whether we have reached the limit is detected by another `ep_do_intercept_request` filter with a higher priority (0). Now, if we need to stop the request, there is no other way for us (or maybe I miss one?) to stop *this* filter, because it unconditionally calls `vip_safe_wp_remote_request()` (this is what we don't need it to do).

Reply:
I'm not a fan of putting this logic into each and every hook onto `ep_do_intercept_request` (especially if we do decide to add more on later). But I understand your point. 

Would it be possible to have a filter that runs the check and then, short-circuit it before even adding the other filters?

Reply:
I'm not a fan of replicating this logic into each and every hook onto ep_do_intercept_request. But I understand your point. I'm going to combine these filters into one.

Reply:
@sjinks I've opened https://github.com/Automattic/vip-go-mu-plugins/pull/2712.